tests/always/test_example_dags.py

```diff
@@ -60,7 +60,7 @@ def test_should_be_importable(example):
 
 @pytest.mark.parametrize("example", example_dags_except_db_exception(), ids=relative_path)
 def test_should_not_do_database_queries(example):
-    with assert_queries_count(0):
+    with assert_queries_count(0, ignore_patterns=['dagbag.py:validate_task_pools']):
```

Comment:
Are you sure we want to do it @dstandish  ? This basically means that whenever we parse DAGs we make Database queries.

Reply:
valid concern.  maybe it's possible to query once at start of dag processing and let all the processes have the information through an attr.

Reply:
wait.... nevermind... as it is, we only make _one_ query at the end after all DAG parsing has completed.  so it's not like it's querying once for every DAG.  this to me seems pretty reasonable? 

but I guess DagBag load doesn't only happen from the dag processor service, but also perhaps when tasks are run....  so maybe there's something here to optimize.

perhaps we could make it so that this kind of check is disabled except when loading dagbag from dag processor service. WDYT?

Reply:
hmm moving this to dag processor seems like generally the right direction but i think i may need to rethink the sync logic because we won't always have all warnings in memory, since the processor may just run for a specific dag file and not the entire dags folder...

Reply:
ok @potiuk i moved the logic to dag processor.  so it will not run in every DagBag load, only when dags are loaded by the dag processor